src/Hyperopt.jl

```diff
@@ -214,31 +214,50 @@ function pmacrobody(ex, params, ho_, pmap=pmap)
             # <https://docs.julialang.org/en/v1/manual/distributed-computing/#Local-invocations>
             put!(ho_channel, deepcopy(ho))
 
-            # We don't care about the results of the `pmap` since we update the hyperoptimizer
-            # inside the loop.
-            $(esc(pmap))(1:ho.iterations) do i
-                # We take the hyperoptimizer out of the channel, and get our new parameter values
-                local_ho = take!(ho_channel)
-                # We use `update_history` because we want to only update the history once we've
-                # finished the iteration and have a result to report back as well. Otherwise,
-                # some processes may observe the Hyperoptimizer in an inconsistent state with
-                # `length(ho.history) > length(ho.results)`. Moreover, if one run is very quick
-                # the history and results could become out of order with respect to one another.
-                $(Expr(:tuple, esc.(params)...)), _ = iterate(local_ho, i; update_history = false)
-                # Now we put it back so another processor can use the hyperoptimizer
-                put!(ho_channel, local_ho)
-
-                # Now run the objective
-                res = $(esc(ex.args[2])) # ex.args[2] = Body of the For loop
-
-                # Now update the results; we again grab the one true hyperoptimizer,
-                # and populate it's history and result.
-                local_ho = take!(ho_channel)
-                push!(local_ho.history, $(Expr(:tuple, esc.(params[2:end])...)))
-                push!(local_ho.results, res)
-                put!(ho_channel, local_ho)
-
-                res
+            # pmap continues running a few lefterover jobs when interrupted, this is to stop that
+            done_channel = RemoteChannel(() -> Channel{Bool}(1), 1)
+            put!(done_channel, false)
+
+            try
+                # We don't care about the results of the `pmap` since we update the hyperoptimizer
+                # inside the loop.
+                $(esc(pmap))(1:ho.iterations) do i
+                    isdone = take!(done_channel)
+                    put!(done_channel, isdone)
+                    isdone && return
+
```

Comment:
I think I realised why my tests behaved different from what I expected. 

When generating my errors/interrupts I do them in the code block run on the workers, so the exception will happen on the worker.
When I press ctrl-C to interrupt jobs on my cluster I do that on the main process so it is caught by the try statement.

It seems that an exception on a worker can generate an exception back to the main process, but maybe not the same? Currently I have 
```
Worker 2 terminated.
Interrupt handling: Error During Test at /home/runner/work/Hyperopt.jl/Hyperopt.jl/test/runtests.jl:109
  Got exception outside of a @test
  On worker 2:
  InterruptException:
...
```
which seems like it does generate the expected interrupt exception on the worker, but the worker can't handle it and stops which generates an error on the main process? I think this might be how you'd want it to work, it just makes it harder to test.